### PR TITLE
blog post to automate mysql backups to CF

### PR DESCRIPTION
**Pull request by wbentley15**
Bios:
Walter Bentley – Rackspace Private Cloud Solutions Architect – Walter is a new Racker with a diverse background in Production Systems Administration and Solutions Architecture. He brings over 15 years of experience across numerous industries such as Online Marketing, Financial, Insurance, Aviation, Food Industry and Education. In the past, always being the requester, consumer and advisor to companies to use technologies such as OpenStack, now promoter of OpenStack technology and Cloud educator.

Twitter: @djstayflypro
LinkedIn: http://goo.gl/r2p21i
Github: wbentley15

Accept to allow redaction, editing or structure changes by technical writer staff desired date of publishing
Publish date: 02-17-2015

### CODE REVIEW

**Comment by wbentley15** [comment]
@catlook as always...I need your special 'cat eyes' on this.  Thank you in advance for your help!

**Comment by rgbkrk** [comment]
Found the cat eyes @catlook

**Comment by catlook** [comment]
On it!

-- Cat Lookabaugh, Information Developer
Rackspace Hosting – the #1 Managed Cloud Company
Developer Relations Group
Office: (210) 312-8193   Cell: (210) 364-1060
cat.lookabaugh@rackspace.com

We should not write so that it is possible for the reader to understand us, but so that it is impossible for him to misunderstand us. - Quintilian, rhetorician (c. 35-100)

Harmony + Learner + Responsibility + Context + Intellection + Arranger + Consistency + Relator + Input + Analytical

File: src/site_source/_posts/2015-02-17-mysql-backup-to-rackspace-cloud-files.md [hunk 1]
```diff
@@ -0,0 +1,65 @@
...
+published: true
+categories:
+    - rackspace-public-cloud
+    - rackspace-cloud-files
+    - Ansible 
+---
+
+While this blog post may seen trivial on the surface, it does pack some very interesting information on how very flexible the Rackspace Cloud Files product can be.  While executing another customer project, the age old question of: “Where are we going to put the database backups?” was raised.  Back in the day this question only really had one solution.  In the current age of the cloud, you have a few options.  Since I like to live life on the edge…I raised my hand and said Cloud Files.
+
+For those of you not familiar with Cloud Files, the easiest way to describe it is shared Object Storage.  In OpenStack lingo, you could also call it shared Swift.  Cloud Files is an API enable Object storage capability found on the Rackspace Public cloud platform.  In this post we will walk you thru how easy it is to store something as simple as database backups in Cloud Files using simple automation, fronted by Ansible of course (my orchestration drug of choice).  I promise this post will be short and sweet.
```
Comment (catlook) [inline src/site_source/_posts/2015-02-17-mysql-backup-to-rackspace-cloud-files.md]: ... API enabled Object ...
... In this post, we will ...

File: src/site_source/_posts/2015-02-17-mysql-backup-to-rackspace-cloud-files.md [hunk 2]
```diff
@@ -0,0 +1,65 @@
...
+
+The module we will be using for this exercise is the following:
+
+	rax_files_objects - Upload, download, and delete objects in Rackspace Cloud Files
+
+Disclaimer:  this example assumes you have already created a Cloud Files container but, if not refer to step #2b.
+
+Let’s get started
```
Comment (catlook) [inline src/site_source/_posts/2015-02-17-mysql-backup-to-rackspace-cloud-files.md]: Let’s get started!

File: src/site_source/_posts/2015-02-17-mysql-backup-to-rackspace-cloud-files.md [hunk 3]
```diff
@@ -0,0 +1,65 @@
...
+---
+#####Step #1: Clone Repo
+
+Clone the repo below to pull down the roles you will need.  Consider this a working example that you can later alter for your specific needs.
+
+	$ git clone --recursive https://github.com/wbentley15/rax-ansible-cf.git
+	
+#####Step #2: Examine roles and populate variables
+
+Take a look at the roles and familiarize yourself with the steps.  All the variables you will need to supply values for can be found below and are located in the group_vars directory within the localhost file:
```
Comment (catlook) [inline src/site_source/_posts/2015-02-17-mysql-backup-to-rackspace-cloud-files.md]: ...  the steps. Find below all the variables for which you will need to supply values. The variables are located in the group_vars directory within the localhost file:

File: src/site_source/_posts/2015-02-17-mysql-backup-to-rackspace-cloud-files.md [hunk 4]
```diff
@@ -0,0 +1,65 @@
...
+	raxkey:  Rackspace API key
+	raxuser:  Rackspace Username
+	raxregion:  Rackspace region
+	fileloc:  location of files to upload to Rackspace Cloud Files
+
+#####Step #2b: Create Cloud Files container
+
+You will find two additional roles that you can use to create a Cloud Files container and add files in an ad-hoc manner.
+
+The roles are named below and they are very self explanatory.
```
Comment (catlook) [inline src/site_source/_posts/2015-02-17-mysql-backup-to-rackspace-cloud-files.md]: ... below, and they  ...

File: src/site_source/_posts/2015-02-17-mysql-backup-to-rackspace-cloud-files.md [hunk 5]
```diff
@@ -0,0 +1,65 @@
...
+#####Step #3: Give it a try
+
+Now it is time to give it a try.  Do not forget to edit the `/rax-ansible-cf/group_vars/localhost` file and add hosts to the ‘hosts’ file located in the root directory of the playbook.
+
+Assuming you already have Ansible installed, you can execute the playbook:
+
+	$ cd ~/rax-ansible-cf/
+	$ ansible-playbook -i hosts base-db-backup.yml
+
+Once completed, you will be able to log into the Cloud Control Panel and under the Cloud Files tab find your backups within the container designated.
```
Comment (catlook) [inline src/site_source/_posts/2015-02-17-mysql-backup-to-rackspace-cloud-files.md]: ... and, under the Cloud Files tab, find ...

[Nice article!!]

**Comment by wbentley15** [comment]
@catlook thanks again!  @rgbkrk all done with edits and ready to go.

**Comment by rgbkrk** [comment]
Let's do this!

**Pull request merged**